File: top-interview-questions-easy/dynamic programming/best-time-to-buy-and-sell-stock/121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let max = 0
  for (let i = 0; i < prices.length; i++) {
    max = Math.max(max, Math.max(...prices.slice(i)) - prices[i])
  }
  return max
}

File: top-interview-questions-easy/dynamic programming/best-time-to-buy-and-sell-stock/121.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        max_profile, min_profile = 0, float('inf')
        for price in prices:
            min_profile = min(min_profile, price)
            max_profile = max(max_profile, price - min_profile)
        return max_profile
